Docs 6036 add message looging policy example pa (#305)
mulesoft/docs-api-manager: modules/ROOT/pages/message-logging-policy.adoc

|After Calling API |Set to cause your logging policy to log messages after the API endpoint is called, taking into account the order in which this policy is applied |Optional
|===

> == Example of Logging Policy from the UI

Reviewer: Why do we need to say "from the UI"?

Author: Because we have one for XML and one from the UI.

> To better understand how to use the Message Logging policy, consider an example "Les Vetments" company that is in the business of retail clothing and more. To run their e-commerce portal successfully, Les Vetments needs to audit every event and transaction made on their website.

Reviewer: english is not my strength, but maybe it's consider as **an** example?

Author: Leaving it to my editor. This is how she suggests we write though.

For example, if a customers buys or returns a product, or if the company adds or removes a product from the catalogue, the transactions must be logged. The IT manager at Les Vetments applies the Message Logging policy to their APIs to enable logging and viewing any data or transaction changes without making any modifications in the code.

> Les Vetments uses the resource-level policy's granularity to apply the policy to only `POST`, `PUT`, `PATCH` and `DELETE` methods. To log all events, Les Vetments applies the Message Logging policy to the API with the following UI configurations:

Reviewer: The resource level is needed IF we say that the requirement is only to log teh events that make changes to their data. If not, we can remove the resource level configuration.

Author: Done! Thanks!

> *Message*: #[payload]

Reviewer: Correction here, the expression to use is #[message], and everything in the message will be logged. We can then add the following examples to show other alternatives:

#[attributes.headers] if the requirement is to log the headers
OR
#['User ' ++ authentication.clientId ++ ' performed action ' ++ attributes.method ++ ' on ' ++ attributes.requestPath ++ ' with Payload: ' ++ payload]
It's an example where there is more customization in the information to log, in this case you see the user, action being performed and the body.

Author: Done! Thanks!

+
Everything in the payload must be logged.
*Conditional*: Blank
+

> Because Les Vetments requires all types of messages, no Mule expression ito filter messages is specified.

Reviewer: typo in ito

Author: Done! Thanks!

*Category*: Blank
+
No prefix is required in the log sentence.
*Level*: Info
+
All types of messages are to be logged: Info, Warn, Error, or Debug.
*Before Calling API*: checked
+
All transactions occurring before the API is called is to be logged.
*After calling API*: Checked
+
All transactions occurring after the API is called is to be logged.

Now, whenever a customer accesses the Les Vetments catalogue and adds or removes items from their cart, or when an item is added or removed from the catalogue, logs are generated:

> //add the output example here

Reviewer: Can you try again creating an API from scratch and applying the policy with this configuration to see if you can see the output now?

Author: Done! Thanks!

|After Calling API |Set to cause your logging policy to log messages after the API endpoint is called, taking into account the order in which this policy is applied |Optional
|===

== Example of Message Logging Policy Configured from the UI

> To better understand how to use the Message Logging policy, consider the "Les Vetments" company that is in the business of retail clothing and more. To run their e-commerce portal successfully, Les Vetments needs to audit the status code of the requests made on their website.

Reviewer 2: Does your audience really need "and more"?
It is vague and doesn't add value to the scenario, unless you mention something besides clothing in it.

Reviewer 2: ```suggestion
To better understand how to use the Message Logging policy, consider a fictional retail clothier named Les Vetments. To successfully run its e-commerce portal, Les Vetments must audit the status code of the requests made on its website.
```

Author: Done! Thanks!

> For example, if a customers buys or returns a product, or if the company adds or removes a product from the catalogue, the transactions must be logged. The IT manager at Les Vetments applies the Message Logging policy to their APIs to enable logging and viewing any data or transaction changes without making any modifications to the code.

Reviewer 2: ```suggestion
For example, if a customers buys or returns a product, or if the company adds or removes a product from its catalog, the transactions must be logged. The IT manager at Les Vetments applies the Message Logging policy to the company APIs to enable logging and viewing any data or transaction changes without making any modifications to the code.
```

Author: Done! Thanks!

> To log the attributes of the incoming and outgoing HTTP Message, Les Vetments applies the Message Logging policy to the API with the following UI configurations:

Reviewer 2: ```suggestion
To log the attributes of the incoming and outgoing HTTP message, Les Vetments applies the Message Logging policy to the API with the following UI configurations:
```

Author: Done! Thanks!

> To log status code of the API response that is being returned to the calling client application, Les Vetments applies the Message Logging policy to the API with the following UI configurations:

Reviewer: I've discussed it with Fede, he brings a good point that we want to avoid logging the payload. So the proposal is to log the message attributes, this is the expression #[attributes], here there is extra doc about how a messaged is composed and the attributes: https://docs.mulesoft.com/mule-runtime/4.3/intro-mule-message.

Author: Done! Thanks!

> To log the status code of the API response that is being returned to the calling client application, Les Vetments applies the Message Logging policy to the API with the following UI configurations:

Reviewer 3: To log the attributes of the incoming and outgoing HTTP Message, Les Vetments applies the Message Logging policy to the API with the following UI configurations:

Author: Done! Thanks!

> * *Message*: #[“Status code of the response is: ” ++ attributes.statusCode]

Reviewer 3: Update the Message to #["Current HTTP Attributes are: " ++ attributes]

Author: Done! Thanks!

The attributes in the payload must be logged.
* *Conditional*: Blank

Because Les Vetments requires all attributes, no Mule expression to filter messages is specified.
* *Category*: Blank

No prefix is required in the log sentence.
* *Level*: Info

> All types of messages are to be logged: Info, Warn, Error, or Debug.

Reviewer 2: Do we need to explain why or clarify further that specifying Info also includes Warn, Error, and Debug?

Author: Done! Thanks!

> * *Before Calling API*: Unchecked

Reviewer: This should be checked

Author: Done! Thanks!

All transactions that occur before the API is called are to be logged.
* *After calling API*: Checked

All transactions that occur after the API is called are to be logged.

> Now, whenever a customer accesses the Les Vetments catalogue, adds or removes items from their cart, or when an item is added or removed from the catalogue, logs are generated:

Reviewer 2: ```suggestion
Following this configuration, whenever customers access the Les Vetments catalogue or add or remove items from their carts, or when an item is added or removed from the Les Vetments catalog, logs are generated:
```

Author: Done! Thanks!
